Modules/_sqlite/connection.c

```diff
@@ -2173,6 +2173,45 @@ pysqlite_connection_create_collation_impl(pysqlite_Connection *self,
     Py_RETURN_NONE;
 }
 
+/*[clinic input]
+_sqlite3.Connection.set_file_control as pysqlite_connection_set_file_control
+
+    op: int
+      a SQLITE_FCNTL_ constant
+    arg: long
+      argument to pass
+    /
+    dbname: str = NULL
+      database name
+
+Invoke a file control method on the database.
+[clinic start generated code]*/
+
+static PyObject *
+pysqlite_connection_set_file_control_impl(pysqlite_Connection *self, int op,
+                                          long arg, const char *dbname)
+/*[clinic end generated code: output=d9d2d311892893b6 input=0253798d9514fea2]*/
+{
+    int rc;
+    long val = arg;
```

Comment:
Not all `sqlite3_file_control` opcodes use the `int`[^1] datatype. For example `SQLITE_FCNTL_VFSNAME` and `SQLITE_FCNTL_TEMPFILENAME` are `char *`. Suggesting to limit this (like we do for `sqlite3_db_config`) to only the supported opcodes, and raise an exception for unsupported opcodes.

In any case, we should use `int` here.

```suggestion
    int val = arg;
```

[^1]: `int`, not `long`

Reply:
Please also address the first part of this remark.

Reply:
oops! sorry for making you ask twice

Reply:
Done. I have also removed the SQLITE_FCNTL_TEMPFILENAME and SQLITE_FCNTL_VFSNAME constants. I hope that's alright.